File: Python3.x/66-Plus One.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        digits[-1] += 1
        for i in range(len(digits) - 1, -1, -1):
            if digits[i] > 9:
                digits[i] -= 10
                if i > 0:
                    digits[i - 1] += 1
                else:
                    digits = [1] + digits
        return digits


class Solution:
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        if len(digits) == 0:
            digits = [1]
        elif digits[-1] == 9:
            digits = self.plusOne(digits[:-1])
            digits.extend([0])
        else:
            digits[-1] += 1
        return digits

File: Algorithms/C++/66-Plus_One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        (*(digits.end() - 1))++;
        for (int i = digits.size() - 1; i >= 0; --i) {
            if (digits[i] > 9) {
                digits[i] -= 10;
                if (i > 0) {
                    digits[i - 1]++;
                } else {
                    digits.insert(digits.begin(), 1);
                }
            }
        }
        return digits;
    }
};
